### PR TITLE
[v6r22] Pilot wrapper for pilot3: download files at runtime

### PR DESCRIPTION
**Pull request by fstagni**
I modified the pilot wrapper so that it now makes different things depending if it's pilot 2 (still the default) or pilot 3. For pilot 2 nothing changes. For pilot 3 the input files (.py and .json) will be downloaded at run time. This may be improved with time.

@atsareg : VMDIRAC should maybe invoke the method `pilotWrapperScript` in the PilotWrapper.py module to get pilot wrapper to be used there. I think what's used in VMDIRAC is https://github.com/DIRACGrid/VMDIRAC/blob/integration/VMDIRAC/WorkloadManagementSystem/Bootstrap/vm-pilot and IIUC it's pilot 2.

BEGINRELEASENOTES

*WMS
CHANGE: Pilot wrapper for pilot3: download files at runtime

ENDRELEASENOTES

### CODE REVIEW

File: WorkloadManagementSystem/Agent/SiteDirector.py [hunk 1]
```diff
@@ -829,8 +824,7 @@ def _submitPilotsToQueue(self, pilotsToSubmit, ce, queue):
                   stampDict is a dict of timestamps of pilots submission
        :rtype: dict
     """
-    self.log.info('Going to submit a maximum of %d pilots to %s queue' %
-                  (pilotsToSubmit, queue))
+    self.log.info('Going to submit a maximum of %d pilots to %s queue' % (pilotsToSubmit, queue))
```
Comment (chaen) [inline WorkloadManagementSystem/Agent/SiteDirector.py]: Please split the log message

File: WorkloadManagementSystem/Utilities/PilotWrapper.py [hunk 2]
```diff
@@ -15,21 +14,60 @@
...
-import json
 import base64
 import bz2
+import logging
+import time
+import urllib2
+import tarfile
 
-from cStringIO import StringIO
+# setting up the logging
```
Comment (chaen) [inline WorkloadManagementSystem/Utilities/PilotWrapper.py]: These scripts in string are quite hard to read and are error prone, wouldn't it be better to put them in separate files ?

File: WorkloadManagementSystem/Agent/SiteDirector.py [hunk 3]
```diff
@@ -507,8 +503,8 @@ def submitJobs(self):
 
       ce, ceDict = self._getCE(queueName)
 
-      pilotsWeMayWantToSubmit, additionalInfo = self._getPilotsWeMayWantToSubmit(
-          ceDict)  # additionalInfo is normally taskQueueDict
+      # additionalInfo is normally taskQueueDict
+      pilotsWeMayWantToSubmit, additionalInfo = self._getPilotsWeMayWantToSubmit(ceDict)
       self.log.verbose('%d pilotsWeMayWantToSubmit are eligible for %s queue' % (pilotsWeMayWantToSubmit, queueName))
```
Comment (chaen) [inline WorkloadManagementSystem/Agent/SiteDirector.py]: Please split the log message (or use debug)

File: WorkloadManagementSystem/Agent/SiteDirector.py [hunk 4]
```diff
@@ -507,8 +503,8 @@ def submitJobs(self):
...
       if not pilotsWeMayWantToSubmit:
         self.log.verbose('...so skipping %s' % queueName)
```
Comment (chaen) [inline WorkloadManagementSystem/Agent/SiteDirector.py]: Please split the log message (or use debug)

File: WorkloadManagementSystem/Agent/SiteDirector.py [hunk 5]
```diff
@@ -583,7 +578,7 @@ def submitJobs(self):
         # updating the pilotAgentsDB... done by default but maybe not strictly necessary
         res = self._addPilotTQReference(queueName, additionalInfo, pilotList, stampDict)
 
-    self.log.info("%d pilots submitted in total in this cycle," % self.totalSubmittedPilots)
+    self.log.info("%d pilots submitted in total in this cycle" % self.totalSubmittedPilots)
```
Comment (chaen) [inline WorkloadManagementSystem/Agent/SiteDirector.py]: Please split the log message

File: WorkloadManagementSystem/Utilities/PilotWrapper.py (hunk 2 above)
Comment (fstagni) [reply WorkloadManagementSystem/Utilities/PilotWrapper.py]: I understand, but this script (the pilot wrapper) is composed on the fly, it makes not much sense to put it in a separate file.

File: WorkloadManagementSystem/Agent/SiteDirector.py [hunk 6]
```diff
@@ -491,7 +487,7 @@ def submitJobs(self):
 
     for queueName, queueDictionary in queueDictItems:
       # now submitting to the single queues
-      self.log.verbose("Evaluating queue %s" % queueName)
+      self.log.verbose("Evaluating queue", "%s" % queueName)
```
Comment (atsareg) [inline WorkloadManagementSystem/Agent/SiteDirector.py]: Just queueName as the last argument ?

**Pull request merged**